src/lib/GraphQL/Schema/MatrixFieldDefinitionMapper.php

```diff
@@ -23,11 +25,27 @@ class MatrixFieldDefinitionMapper extends DecoratingFieldDefinitionMapper implem
     /** @var \Ibexa\Contracts\Core\Repository\ContentTypeService */
     private $contentTypeService;
 
-    public function __construct(FieldDefinitionMapper $innerMapper, NameHelper $nameHelper, ContentTypeService $contentTypeService)
-    {
+    /* @var iterable<\Ibexa\FieldTypeMatrix\FieldType\Mapper\FieldTypeToContentTypeStrategyInterface> */
+    private iterable $strategies;
+
+    /**
+     * @param iterable<\Ibexa\FieldTypeMatrix\FieldType\Mapper\FieldTypeToContentTypeStrategyInterface> $strategies
+     */
+    public function __construct(
+        FieldDefinitionMapper $innerMapper,
+        NameHelper $nameHelper,
+        ContentTypeService $contentTypeService,
+        iterable $strategies
+    ) {
         parent::__construct($innerMapper);
         $this->nameHelper = $nameHelper;
         $this->contentTypeService = $contentTypeService;
+
+        foreach ($strategies as $strategy) {
+            if ($strategy instanceof FieldTypeToContentTypeStrategyInterface) {
+                $this->strategies[] = $strategy;
+            }
+        }
```

Comment:
Do not skip non-`FieldTypeToContentTypeStrategyInterface`. Providing an instance of something else is a logical error and should not be allowed.

From my perspective, if you are only ever iterating over this set of strategies (which as far as I can see you are) then you do not need to unpack them. You assign this variable directly.

You do not need to validate that they are the right interfaces, because we can assume (again, logic error) that only right services are provided.

```suggestion
        $this->strategies = $strategies;
```